src/inference/particle_filter.jl

```diff
@@ -111,7 +111,16 @@ end
     particle_filter_step!(state::ParticleFilterState, new_args::Tuple, argdiffs,
         observations::ChoiceMap, proposal::GenerativeFunction, proposal_args::Tuple)
 
-Perform a particle filter update, where the model arguments are adjusted, new observations are added, and some combination of a custom proposal and the model's internal proposal is used for proposing new latent state (whatever is not proposed from the custom proposal will be proposed using the model's internal proposal).
+Perform a particle filter update, where the model arguments are adjusted, new
+observations are added, and some combination of a custom proposal and the
+model's internal proposal is used for proposing new latent state.  That is, for
+each particle, the value of a random choice at address `a` in the new trace for
+that particle is chosen by the following rule:
+* The particle's previous trace must not have a random choice at address `a`:
+  overwriting old state is not allowed.
+* If the proposal's choicemap has a random choice at address `a`, use that
+  value.
+* Otherwise, sample the value from the model's internal proposal.
```

Comment:
@bzinberg Actually, looking at the code, I don't think this happens? In the current code, the internal proposal is not used at all (except, I guess, for untraced randomness) -- the user's proposal is responsible for sampling all new choices. (This is because `update` is used.)

(It's not clear how to relax this restriction without losing incremental computation, in Gen's current GFI. I think a previous version of Gen had more variants of `update`, including ones that sampled "missing" values from the internal proposal.)

Reply:
If `generate` were used [here](https://github.com/probcomp/Gen/blob/efc5a00b787cabf1e26e6849727ae3f10dc44868/src/inference/particle_filter.jl#L131) then the internal proposal would be used to generate missing addresses. But this wouldn't give us an incrementally computed weight.

Reply:
@alex-lew @bzinberg update does use the internal proposal by default to fill in any choices that are not determined by the previous trace or the constraints choice map.

Reply:
@marcoct, in that case, is this proposed doc accurate?

Reply:
@bzinberg In my reading, the proposed doc implies that there are no addresses copied from the previous trace to the new trace.

Reply:
@marcoct Ah, thanks... I think I confused it with custom-proposal MH, which errors if you don't provide every choice -- but that's because of `assess` fails to compute the backward probability, not because `update` fails to sample values for missing choices.